src/Detection/Framework/Astro.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Utopia\Detector\Detection\Framework;
+
+use Utopia\Detector\Detection\Framework;
+
+class Astro extends Framework
+{
+    public function getName(): string
+    {
+        return 'Astro';
+    }
+
+    public function getFiles(): array
+    {
+        return ['astro.config.mjs', 'astro.config.js', '.astro/'];
+    }
+
+    public function getFileExtensions(): array
+    {
+        return ['astro', 'js', 'ts'];
+    }
+
+    public function getLanguages(): array
+    {
+        return ['JavaScript', 'TypeScript', 'Astro'];
+    }
```

Comment:
Extensions and languages detection (compared to looking for specific files) should be a fallback, and VERY rarely needed for sites.
I would recommend to only returning what 100% identifies Astro correctly. Saying JavaScript here or js isn't useful, since all frameworks have those.

```suggestion
    public function getFileExtensions(): array
    {
        return ['astro'];
    }

    public function getLanguages(): array
    {
        return ['Astro'];
    }
```

Reply:
We don't use these methods for framework detection anyway, so I am removing them.